src/targets/gpu/include/migraphx/gpu/device/topk.hpp

```diff
@@ -0,0 +1,32 @@
+#ifndef MIGRAPHX_GUARD_RTGLIB_DEVICE_TOPK_HPP
+#define MIGRAPHX_GUARD_RTGLIB_DEVICE_TOPK_HPP
+
+#include <migraphx/argument.hpp>
+#include <migraphx/config.hpp>
+#include <hip/hip_runtime_api.h>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+namespace gpu {
+namespace device {
+
+argument topk_smallest(hipStream_t stream,
```

Comment:
Wouldn't it be better to pass in `largest` (from `op.largest`) here and consolidate into one `topk()` function? Then in topk.cpp, you can just use the ternary operator on `largest` and call topk with either `std::less` or `std::greater`.

Reply:
This current approach is what Paul proposed, with this approach, we can avoid checking the op.largest every time the compare operator is used.